src/databricks/labs/remorph/reconcile/connectors/oracle.py

```diff
@@ -0,0 +1,68 @@
+from pyspark.sql import DataFrame, DataFrameReader
+
+from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
+
+
+class OracleDataSource(DataSource):
+
+    # TODO need to check schema_name,catalog_name is needed
+    # TODO rename the table_or_query
+    def read_data(self, schema_name: str, catalog_name: str, table_or_query: str,
+                  table_conf: Tables) -> DataFrame:
+        try:
+            if table_conf.jdbc_reader_options is None:
+                return (self.reader(table_or_query)
+                        .options(**self._get_timestamp_options())
+                        .load())
+            else:
+                return (reader
+                        .options(**self._get_jdbc_reader_options(table_conf.jdbc_reader_options) |
+                                   self._get_timestamp_options())
+                        .load())
+        except PySparkException as e:
+            error_msg = (
+                f"An error occurred while fetching Oracle Data using the following {query} in OracleDataSource : {e!s}"
+            )
+            raise PySparkException(error_msg) from e
+
+    def get_schema(self, table_name: str, schema_name: str, catalog_name: str) -> list[Schema]:
+        try:
+            schema_query = self._get_schema_query(table_name, schema_name)
+            schema_df = self.reader(schema_query).load()
+            return [Schema(field.column_name.lower(), field.data_type.lower()) for field in schema_df.collect()]
+        except PySparkException as e:
+            error_msg = (
+                f"An error occurred while fetching Oracle Schema using the following {query} in OracleDataSource: {e!s}"
+            )
+            raise PySparkException(error_msg) from e
+
+    oracle_datatype_mapper = {
```

Comment:
this should be property

Reply:
Can't make this as a property ,as it needs class initialisation.